### DISCUSSION TITLE
Skip CLEAN - how to clean all the Android generated files and start fresh

### DISCUSSION THREAD

**Discussion by davidakoontz** [skiptools/skip#228] [Q&A] [answered]
I'm getting some Xcode Errors like:

Execution failed for task 'app.compileDebugKotlin'.
Unresolved reference 'launch'. in Main.kt
The gradle command failed. Review log for details...

Configuring project ':SkipModel' without an existing directory is deprecated.
Configuring project ':SkipLib' without an existing directory is deprecated.
etc.

I'm searching for a way to CLEAN the generated / derived files in the Android side of the dual platform.

I see notes on cleaning the Xcode (iOS) side of things... but that does NOT clean the Android side.
Didn't find any way to do this properly... should I start deleting folders?

Seems like a Gradle CLEAN command  or a Fastlane CLEAN command would be nice right about HERE.

**Comment by davidakoontz** [comment]
For Example - what happens if I just DELETE the .build (hidden folder)?

**Comment by davidakoontz** [comment]
Here is Xcode complaining about Main.kt  UIApplication.launch(this)

<img width="1387" alt="Screenshot 2024-09-08 at 9 39 31 PM" src="https://github.com/user-attachments/assets/d1564cd0-163b-409a-859f-55d4f7c492d5">

I'm looking thru the skip-grade.log.txt - but nothing seems to indicate what I should do about the issues in the log.

**Reply by aabewhite** [reply, marked as answer]
If you're building from Xcode, then the .build folder shouldn't come into play, but you can delete it if you like. It can't hurt.

To get a truly clean build you can do the same thing you do for iOS: tell Xcode to Clean and also delete DerivedData.

When I see the error pictured above, my first question is whether you have an old version of skip-ui where UIApplication.launch does not exist. Make sure to run File->Packages->Update to Latest Versions to ensure you have the latest libraries.

**Comment by davidakoontz** [comment]
Ah... well a touch different after Update:
<img width="218" alt="Screenshot 2024-09-08 at 10 14 52 PM" src="https://github.com/user-attachments/assets/071ffa93-0cb5-40ff-ba5f-a261944d093f">
<img width="210" alt="Screenshot 2024-09-08 at 10 15 35 PM" src="https://github.com/user-attachments/assets/c8ee14eb-c2b6-46c3-8e82-a9bfbb3bc17f">

Well - I'm always forgetting to Start up Android-Studio & a Pixel Virtual Device...  but after restarting the world ...

Something WORKED.

So just so that I'm clear and not developing superstitions...  You seem to be infering that deleting DeriedData folder is a WAY to Clean the Android side of the Dual project.

I had done the Delete DeviredData folder a few times...  it had not resolved my build issues...  But ALL is well with the World NOW!

Thanks Abe!

**Reply by aabewhite** [reply]
I'd have to check the logs to see when UIApplication.launch was introduced, but I'm guessing that what fixed things here is just getting the latest libraries.

And in general when you're building with Skip from Xcode, the pattern is the same as building without Skip: Generally just rebuilding should work. Occasionally you need to do a clean build because Xcode is caching something it shouldn't. And very occasionally, the only way to get Xcode to stop caching some old bit of data is to delete DerivedData. (Yes, when you build in Xcode, the Android side is being generated and built within DerivedData too)

**Comment by davidakoontz** [comment]
OK - for FUTURE ME...
Fastlane uses the hidden ".build" folder - to clean it just delete the folder  > rm -rf .build

Xcode uses the DerivedData folder - to clean it use Xcode > Product > Clean Build Folder ; AND/OR delete DerivedData.

**Reply by davidakoontz** [reply]
Why use a hidden folder in the first place... try better Names:  e.g.  "fastbuild"  to stick with Fastlane convention of "Fastfile"
